crates/sui-sdk/src/apis.rs

```diff
@@ -558,6 +573,23 @@ impl ReadApi {
         )
     }
 
+    /// Subscribe to a stream of Sui transaction effects
+    pub async fn subscribe_transaction(
+        &self,
+        filter: TransactionFilter,
+    ) -> SuiRpcResult<impl Stream<Item = SuiRpcResult<SuiTransactionBlockEffects>>> {
+        match &self.api.ws {
+            Some(c) => {
+                let subscription: Subscription<SuiTransactionBlockEffects> =
+                    c.subscribe_transaction(filter).await?;
+                Ok(subscription.map(|item| Ok(item?)))
+            }
+            _ => Err(Error::Subscription(
+                "Subscription only supported by WebSocket client.".to_string(),
+            )),
+        }
```

Comment:
nit: preference for early return style here.

```suggestion
        let Some(c) = &self.api.ws else {
            return Err(Error::Subscription(
                "Subscription only supported by WebSocket client.".to_string(),
            ));
        };
        let subscription: Subscription<SuiTransactionBlockEffects> =
            c.subscribe_transaction(filter).await?;
        Ok(subscription.map(|item| Ok(item?)))
```